src/gpu/intel/ocl/micro_sdpa.cpp

```diff
@@ -230,15 +268,15 @@ status_t micro_sdpa_t::pd_t::init_microkernels(impl::engine_t *engine) {
     arch_ = dev_info->gpu_arch();
     auto *d = desc();
 
-    VCONDCHECK(primitive, create, check, sdpa, mayiuse_microkernels(engine),
-            status::unimplemented,
+    VCHECK_SDPA_COND(mayiuse_microkernels(engine),
             "Microkernels not supported by the OpenCL driver.");
 
     /* Retrieve pre-tuned kernel configuration */
     sdpa_config_t *config = nullptr;
     bool thin_q = (d->queries() <= 16);
     bool quantized = with_key_scales() || with_key_zp() || with_value_scales()
             || with_value_zp();
+    bool is_integrated = compute_engine->device_info()->is_integrated();
```

Comment:
The last time @rjoursler tried to make it work for benchdnn there were some ambiguity for some SKUs which didn't help to identify integrated property reliably. Has it change since then, or is there a safe guard to take a pessimist path in undefined situation?

Reply:
I haven't looked into it. I noticed it was being used in gemm so I would assume it worked correctly here.

Reply:
Dima's point is a good one. There was a PR (I think from Roy) that expanded the is_integrated checks to cover more known cases. Currently, is_integrated in GEMM was only there to distinguish ARL from DG2, and the query currently doesn't distinguish BMG vs. LNL.

@umar456 -- let me send you a patch for BMG vs. LNL.

Reply:
@petercad Thanks for the patch! Its working as expected.